community/modules/compute/gke-job-template/variables.tf

```diff
@@ -50,6 +50,12 @@ variable "allocatable_cpu_per_node" {
   default     = [-1]
 }
 
+variable "has_gpu" {
+  description = "Do nodes have GPUs attached. Generally populated from gke-node-pool via `use` field."
```

Comment:
There are a few issues I see here:

1. I believe the point is to produce a job template that require GPUs, not whether nodes have GPUs (even if that is the logic of the Toolkit) so the description should say that.
2. I don't quite like crossing the line into having our module variable descriptions adopt Toolkit terms in part because the behavior of "use" takes some explaining.

Please change to something more like "Indicates that the job should request nodes with GPUs. Matches output value from gke-node-pool."

I think this can develop and be discussed with time, but I think a job or a deployment variable should express the request for GPUs and then node pools should be created to match the job. The approach we're taking here is backwards, in a sense, from how the job submitter thinks.

I also foresee difficulty surrounding `use` given that you appear to be producing a single template yet potentially consuming multiple node pools. The way I read the code, if you produce a set of node pools with at least one having GPUs, then supply that set to this module with `use`, then the job will always request GPUs.

I realize this is a difficult problem but we should probably flesh out a target or maybe reject CPU/GPU of heterogeneity in creating a template. This can develop with time and I'm not requesting changes other than the variable description.

Reply:
Ultimately by "backwards" what I mean is (A) ideally the variable names would be different as you note and (B) the role of a Job (in the YAML k8s sense) is to describe its runtime requirements.

If you take a long-term reproducibility perspective, a node selector on pool name is not a great way of doing that. You'd rather say something like "I need an Intel processor above this generation" or so on. But this is a good practical solution for a module with the aim of introducing someone to running a (successful) job.